src/directory/footer.rs

```diff
@@ -102,6 +105,27 @@ impl Footer {
     pub fn size(&self) -> usize {
         self.versioned_footer.size() as usize + self.meta.len() + COMMON_FOOTER_SIZE
     }
+
+    /// Compares the version found in the footer with the version of tantivy running
+    /// Indexes are incompatible when the major and minor versions are different between the
+    /// current library and serialised index are different
```

Comment:
This is incorrect. We now have two versions in the Footer. 

Tantivy's version is here for debugging purpose. What is important is the index version. Right now only 0 is available.